pkg/controllers/controllers.go

```diff
@@ -90,7 +89,6 @@ func NewControllers(
 		nodeclaimgarbagecollection.NewController(clock, kubeClient, cloudProvider),
 		nodeclaimtermination.NewController(kubeClient, cloudProvider, recorder),
 		nodeclaimdisruption.NewController(clock, kubeClient, cloudProvider),
-		leasegarbagecollection.NewController(kubeClient),
```

Comment:
Do we have metrics in place and monitoring to ensure that when we remove this that we are not leaking node leases now?

Reply:
Added a test to monitor leaked leases - https://github.com/aws/karpenter-provider-aws/pull/6829